src/promiseAllSettled.ts

```diff
@@ -131,3 +131,16 @@ class PromiseAllSettled<K, T> {
 export function promiseAllSettled<K, T>(options: { keys: K[], promiseFactory: ((key: K) => Promise<T>), sequential: boolean }): Promise<T[]> {
     return new PromiseAllSettled(options.keys, options.promiseFactory, options.sequential).run();
 }
+
+/**
+ * Awaits all provided promises, as if by calling {@link #promiseAllSettled} with
+ * keys being the keys of the array,
+ * promiseFactory returning the promise for each key,
+ * and sequential being false.
+ * @param promises An array of promises.
+ * @return an array of results, if all promises resolved
+ * @throws an Error containing details about all rejected promises, if any promise was rejected
+ */
+export function promiseAllSettledParallel<T>(promises: Array<Promise<T>>): Promise<T[]> {
```

Comment:
As a sidenote I think we could also use `Promise.allSettled(promises)` already provided by typescript in node_modules/typescript/lib/lib.es2020.promise.d.ts
As described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled
) I think it does the same

Reply:
The method was not found when I tried to use it. I don't know why.
I agree that we should try to use it when it is available. Which node versions do support it?
We would still need to wrap the call to unwrap the combined response to get an `all`-like result or rejection.

Reply:
Promise.allSettled(promises) was introduced in Node 12.9.0. 
This might actually cause problems as we allow  "node": ">=8.12.0" in package.json in cplace main repository.
Probably better to leave it as is then...

Reply:
See. I'm running node 12.4, that's why I haven't found it. Luckily. :-)

Reply:
Is anyone of us actually running node 8.12.0? That would be a useful compatibility check.

Reply:
Not sure if anybody is still using Node 8.12.0.
In CircleCI we are currently using node 12.17